### PR TITLE
Add enum device class and possible state options for various sensors

### PR DESCRIPTION
**Pull request by krazos**
`SensorDeviceClass.ENUM` and `SensorEntityDescription.options` were added in core-2023.1 to allow sensors to keep track of possible sensor state values:

https://github.com/home-assistant/core/pull/82489

The possible state values are stored as an attribute and are populated in automation/script editor dropdowns to make it easier for users to select state triggers and/or conditions.

This PR adds the `enum` device class and possible state values for the following sensors:

- Drive Mode
- Gear Selector
- Gear Guard Video Mode
- Gear Guard Video Status
- Software OTA - Status
- Power State

I have tracked these sensor values for a few weeks, changed relevant vehicle settings, etc., and included all possible sensor values that I have observed, but if I have missed any, please feel free to update.

**Example Screenshots:**

_Entity Info:_

image

_Automation Editor:_

image

### CODE REVIEW

**Review by bretterer — APPROVED** [review]
This is a great addition!   Thank you!

File: custom_components/rivian/const.py [hunk 1]
```diff
@@ -179,6 +192,12 @@
             name="Gear Selector",
             icon="mdi:car-shift-pattern",
             key=f"{DOMAIN}_dynamics_propulsion_status_prndl",
+            device_class=SensorDeviceClass.ENUM,
+            options=[
+                "Drive",
+                "Park",
+                "Reverse",
```
Comment (tmack8001) [inline custom_components/rivian/const.py:199]: `Neutral` though not a gear I use frequently, I would imagine this gets pushed through as `neutral`.
We should check that and verify with an actual truck/suv.

**Comment by tmack8001** [comment]
@jrgutier @bretterer and @krazos do any of you know what happens when a value is mapped to a `SensorDeviceClass.ENUM,` that doesn't match a given `option` how is that handled. I can see there being other drive modes added in the future "tank turn" 🤣 , or enabled on employee vehicles and not on our consumer vehicles.

From looking quickly at the provided Home Assistant Core PR this will throw an error log, which isn't ideal though would be a way for our users to report these to us over time as the list of enums diverge from the enum list used by Rivian's servers: 
```
raise ValueError(
  f"Sensor {self.entity_id} provides state value '{value}', "
  "which is not in the list of options provided"
)
```

nothing to do here, just wanted to call out what I believe this change will bring as far as the end user experience is concerned.

File: custom_components/rivian/const.py (hunk 1 above)
Comment (krazos) [reply custom_components/rivian/const.py:199]: Before adding the `Neutral` option, the state change to neutral was ignored and the Gear Selector sensor retained its prior state.  I've added `Neutral` to the list of possible state values, and confirmed that it is working locally.  (The native value is `neutral`, but the options list takes into account the capitalized value from the lambda function.)

IMG_1907

**Comment by krazos** [comment]
Confirmed that state values not included in `options` are ignored (i.e., sensor retains previous state), and show up as an error in the log: 

```
2023-02-13 10:37:14.074 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 182, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 330, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 135, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 390, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 566, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 607, in _async_write_ha_state
    state = self._stringify_state(available)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 572, in _stringify_state
    if (state := self.state) is None:
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 568, in state
    raise ValueError(
ValueError: Sensor sensor.rivian_dynamics_propulsion_status_prndl provides state value 'Neutral', which is not in the list of options provided
```

**Comment by tmack8001** [comment]
> log spamming

It will, but will also be consolidated together given the stack_traces are the same. One of the logs will show `12345 occurrences` similar to what happened when authentication was failing due to lock outs.

I agree that the value to end user is higher than the burden to keep up with future potential changes, not like new drive modes or gears are added regularly 🚀

**Pull request merged**